File: Modules/StateMachine/src/StateMachineStateInterface.cpp
#include <StateMachineStateInterface.h>

namespace DogGE{
    namespace StateMachine{
        void StateMachineStateInterface::setStateMachineId(unsigned int id){
            this->mStateMachineId = id;
        }
        unsigned int StateMachineStateInterface::getStateMachineId(){
            return this->mStateMachineId;
        }
    }
}

File: Modules/StateMachineCs/src/StateMachineStateInterface.cs
namespace StateMachineCs
{
    public abstract class StateMachineStateInterface{
        protected uint mStateMachineId;
        public void setStateMachineId(uint id){
            this.mStateMachineId = id;
        }
        public uint getStateMachineId(){
            return this.mStateMachineId;
        }
        public abstract void enterState();
        public abstract void leaveState();
        public abstract void updateState();
    }
}